tools/agent_tool/src/cpp/main.cpp

```diff
@@ -0,0 +1,245 @@
+                }
+                else if (strcmp(opt.arg, REPEATER_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::REPEATER;
+                }
+                else
+                {
+                    std::cerr << "ERROR: incorrect entity type. Only <client|server|repeater> accepted." << std::endl;
+                    return 1;
+                }
+                break;
+
+            case optionIndex::CONNECTION_ADDRESS:
+                connection_address = std::string(opt.arg);
+                break;
+
+            case optionIndex::LISTENING_ADDRESS:
+                listening_address = std::string(opt.arg);
+                break;
+
+            case optionIndex::NAME:
+                strcpy(name, std::string(opt.arg).c_str());
+                break;
+
+            case optionIndex::CONNECTION_PORT:
+                connection_port = strtol(opt.arg, nullptr, 10);
+                break;
+
+            case optionIndex::LISTENING_PORT:
+                listening_port = strtol(opt.arg, nullptr, 10);
+                break;
+
+            case optionIndex::DOMAIN_ID:
+                domain = strtol(opt.arg, nullptr, 10);
+                break;
+
+            case optionIndex::TRANSPORT:
+                if (strcmp(opt.arg, "tcp") == 0)
+                {
+                    transport_protocol = eprosima::ddsrouter::core::types::TransportProtocol::tcp;
+                }
+                else if (strcmp(opt.arg, "udp") == 0)
+                {
+                    transport_protocol = eprosima::ddsrouter::core::types::TransportProtocol::udp;
+                }
+                break;
+
+            case optionIndex::UNKNOWN_OPT:
+                std::cerr << "ERROR: " << opt.name << " is not a valid argument." << std::endl;
+                option::printUsage(fwrite, stdout, usage, columns);
+                return 1;
+                break;
+        }
+    }
+
+    {
+        if (entity_type == CLIENT)
+        {
+            ip = connection_address;
+            port = connection_port;
+            logUser(AMLIPCPP_MANUAL_TEST, "Starting Manual Agent Node execution. Creating connection address...");
+
+        }
+        else
+        {
+            ip = listening_address;
+            port = listening_port;
+            logUser(AMLIPCPP_MANUAL_TEST, "Starting Manual Agent Node execution. Creating listening address...");
+        }
+
+        // Create connection address
+        auto address = eprosima::ddsrouter::core::types::Address(
+            ip,
+            port,
+            port,
+            transport_protocol);
+
+        try
+        {
+            switch (entity_type)
+            {
+                case CLIENT:
+                {
+                    logUser(AMLIPCPP_MANUAL_TEST, "Address to connect: " << address << ". Creating Node...");
+
+                    // Create Client Node
+                    eprosima::amlip::node::agent::ClientNode client_node(
```

Comment:
The scope of this variable is until the switch clase is closed. Thus, the node is created and destroyed at the moment.
Use a ptr to hold the object